### ISSUE TITLE
SAM transform is only applied by cfn-lint if not declared as a list

### ISSUE THREAD

**Issue by boonew2** [aws-cloudformation/cfn-python-lint#1033]
*cfn-lint version: (`cfn-lint --version`)*
0.22.0
*Description of issue.*
cfn-lint works as expected if the SAM transform is specified like this:
```
Transform: AWS::Serverless-2016-10-31
```
but if multiple transforms are needed (or just declared as a list) like:
```
Transform: 
  - AWS::Serverless-2016-10-31
```
the template is not transformed before the validation occurs and that causes an assortment of false positives.
Please provide as much information as possible:
Working Example:
```
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31
Globals:
  Function:
    Timeout: 3
    Runtime: nodejs8.10
    Environment:
      Variables:
        VARIABLE: VALUE

Resources:
  Function:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: src/
      Handler: index.handler
      Events:
        API:
          Type: Api
          Properties:
            Path: /
            Method: GET
```
Example that yells about `E1001 Top level item Globals isn't valid`:
```
AWSTemplateFormatVersion: '2010-09-09'
Transform: 
  - AWS::Serverless-2016-10-31
Globals:
  Function:
    Timeout: 3
    Runtime: nodejs8.10
    Environment:
      Variables:
        VARIABLE: VALUE

Resources:
  Function:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: src/
      Handler: index.handler
      Events:
        API:
          Type: Api
          Properties:
            Path: /
            Method: GET
```

Cfn-lint uses the [CloudFormation Resource Specifications](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-resource-specification.html) as the base to do validation. These files are included as part of the application version. Please update to the latest version of `cfn-lint` or update the spec files manually (`cfn-lint -u`)

**Comment by kddejong** [comment]
Thanks for submitting this.  Working on it.

**Comment by boonew2** [comment]
@kddejong 
Was the decision to not do the SAM transform at all if multiple transforms were specified just because it didn't seem like the template would be in a knowable state without mystery transforms  being applied? 

The specific use case i had was a transform running before after SAM to paper over a limitation in it ( https://github.com/awslabs/serverless-application-model/issues/717 ):
```
AWSTemplateFormatVersion: 2010-09-09
Transform:
  - PreSamTransformV1
  - AWS::Serverless-2016-10-31
  - PatchSamTransformV1
```

So for that specifically it would definitely work better to always run the SAM transform if there and i imagine that would have a better chance of working in the long run than the almost guaranteed errors that would be thrown by not doing the transform.

Are there plans to allow for custom transform hooks?

**Comment by kddejong** [comment]
Yea you are probably right there.  You want to submit a PR?

**Comment by boonew2** [comment]
Sure! I'll try to churn that out this afternoon!
